tsduck: 3.40-4165 -> 3.44-4676; build fixes

siriobalmelli opened the pull request:
Bump tsduck from 3.40-4165 (master) to 3.44-4676 (latest stable, released May 2026).

Split into three atomic commits:

**1. `tsduck: 3.40-4165 -> 3.42-4421; build fixes`**
- Intermediate bump to 3.42-4421.
- Broaden ignored warnings so newer compilers (clang 17+, gcc 13+) build.
- Extend `tests.patch` to skip additional sandbox-incompatible unit tests.

**2. `tsduck: 3.42-4421 -> 3.44-4676`**
- Bump to 3.44-4676; modern `fetchFromGitHub` (`tag`, `hash`).
- Add `passthru.tests.version` (testers.testVersion) and `passthru.updateScript` (nix-update-script).
- Drop `utimes-fix.patch` (upstreamed in 3.44).
- Add `no-werror.patch` (replaces the previous `NIX_CFLAGS_COMPILE` approach; survives `-Werror` re-additions in upstream sub-makefiles).
- Rewrite `tests.patch` as `tests-sandbox.patch`: remove only the `TSUNIT_DEFINE_TEST` blocks of sandbox-incompatible tests; leave matching `TSUNIT_DECLARE_TEST` declarations in place as inert method declarations. Within-body upstream API changes (e.g. `WebRequest(rep)` -> `WebRequest(&rep)` in 3.44) no longer break patch context, which was the failure mode of the previous monolithic patch.

**3. `tsduck: fall back to SCHED_OTHER`**
- Add `sched-policy-fallback.patch`: ofborg Linux builders run under scheduler policies that `pthread_attr_setschedpolicy` rejects, which previously failed every thread-dependent test before it could start. Retry with `SCHED_OTHER` on `EINVAL`/`EPERM`.

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [N/A] NixOS tests
  - [x] Package tests at `passthru.tests` (`passthru.tests.version`)
  - [N/A] Tests in lib/tests or pkgs/test
- [x] Ran `nixpkgs-review` on this PR
- [x] Tested basic functionality of all binary files
- [x] Fits CONTRIBUTING.md

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `1ee526e83170e9a2fd6c877c36ff90066e99fa45`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `1ee526e83170e9a2fd6c877c36ff90066e99fa45`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `1ee526e83170e9a2fd6c877c36ff90066e99fa45`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `f30f305b54dd9dc0aa80467e2158594866870d59`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

siriobalmelli committed:
tsduck: 3.40-4165 -> 3.42-4421; build fixes

Broaden the range of ignored warnings in order to pass build
on newer compiler versions.

Remove tests which can fail sandbox.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
tsduck: 3.40-4165 -> 3.42-4421; build fixes

Broaden the range of ignored warnings in order to pass build
on newer compiler versions.

Remove tests which can fail sandbox.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
tsduck: 3.42-4421 -> 3.43-4549

Add utimes-fix.patch for missing <sys/time.h> on Darwin
(fixed upstream, drop after next release).

Modernize src fetch to use tag/hash.

Add passthru.tests.version and updateScript.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli committed:
tsduck: 3.42-4421 -> 3.43-4549

Add utimes-fix.patch for missing <sys/time.h> on Darwin
(fixed upstream, drop after next release).

Modernize src fetch to use tag/hash.

Add passthru.tests.version and updateScript.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

siriobalmelli commented:
Bumped to 3.43-4549 (latest stable) and added:
- `passthru.tests.version` — Hydra-friendly version test ✅
- `passthru.updateScript` — future bumps via `nix-update` / r-ryantm
- `utimes-fix.patch` — Darwin build fix (missing `<sys/time.h>`, upstream-fixed on master)

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `9eedc1bddffe27e3e29fc8ceca912cdd81f9fed1`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

iamanaws commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454832`
Commit: [`86078c56a06bbf6f569d4296a7627fa6640bdaef`](https://github.com/NixOS/nixpkgs/commit/86078c56a06bbf6f569d4296a7627fa6640bdaef) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/86078c56a06bbf6f569d4296a7627fa6640bdaef..pull/454832/head))
Merge: [`32816481a6bfbc84e86f92ca9de53e4d21e79b9f`](https://github.com/NixOS/nixpkgs/commit/32816481a6bfbc84e86f92ca9de53e4d21e79b9f)

Logs: https://github.com/iamanaws/nixpkgs-review-gha/actions/runs/23992238092


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

iamanaws reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

siriobalmelli commented on pkgs/by-name/ts/tsduck/package.nix:
```diff
@@ -69,8 +70,19 @@ stdenv.mkDerivation (finalAttrs: {
+  # when the compiler introduces new warnings.
+  # We are packaging release tags, not building and debugging:
+  # if the maintainer has chosen to release it, our job is just to build it.
+  NIX_CFLAGS_COMPILE = [ "-Wno-error" ];
```
fair point, respinning

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `4d7b59d8905a9eb0793e9ffc71abf5c4e7cf1acc`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

SuperSandro2000 commented:
The tests are failing:

```
buildPhase completed in 7 minutes 21 seconds
Running phase: checkPhase
check flags: -j24 SHELL=/nix/store/v8sa6r6q037ihghxfbwzjj4p59v2x0pv-bash-5.3p9/bin/bash NODEKTEC=1 NOGITHUB=1 NOHIDES=1 NOPCSC=1 NOVATEK=1 SYSPREFIX=/ SYSROOT=/nix/store/sfd2ffwiz02c486mbb5v0cv2m5h0y1ha-tsduck-3.43-4549 VERBOSE=y test
\
TSPLUGINS_PATH=/build/source/bin/release-x86_64-localhost LD_LIBRARY_PATH=/build/source/bin/release-x86_64-localhost \
TS_CURL_RETRY="RETRY=5,INTERVAL=100,HOST=tsduck.io" \
/build/source/bin/release-x86_64-localhost/utest 
*** MemoryPluginTest::All, incorrect value, utestMemoryPlugin.cpp, line 209
expected: 3 ("REF_PACKETS_COUNT")
actual:   0 ("output_packets.size()")
*** MessageQueueTest::Queue, assertion failure, utestMessageQueue.cpp, line 145
condition: thread.start()
*** ThreadTest::DeleteWhenTerminated, assertion failure, utestThread.cpp, line 190
condition: thread->start()
*** ThreadTest::MutexTimeout, assertion failure, utestThread.cpp, line 258
condition: thread.start()
*** ThreadTest::Termination, assertion failure, utestThread.cpp, line 141
condition: thread.start()
*** TSProcessorTest::Processing, incorrect value, utestTSProcessor.cpp, line 230
expected: 5 ("5")
actual:   1 ("handler1.logs.size()")
*** FAILURES (6 tests FAILED, 682 passed, 6 assertions FAILED, 32561 passed)
make[2]: *** [Makefile:45: test-shared] Error 1
make[1]: *** [Makefile:42: test] Error 2
make: *** [Makefile:29: test] Error 2
error: Cannot build '/nix/store/hh3i55xk8fqm2pl03nmv7n6jfswradj8-tsduck-3.43-4549.drv'.
```

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `42074b672d00e6087fb675473c8c29cda18133cb`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

siriobalmelli commented:
> The tests are failing:
> 
> ```
> buildPhase completed in 7 minutes 21 seconds
> Running phase: checkPhase
> check flags: -j24 SHELL=/nix/store/v8sa6r6q037ihghxfbwzjj4p59v2x0pv-bash-5.3p9/bin/bash NODEKTEC=1 NOGITHUB=1 NOHIDES=1 NOPCSC=1 NOVATEK=1 SYSPREFIX=/ SYSROOT=/nix/store/sfd2ffwiz02c486mbb5v0cv2m5h0y1ha-tsduck-3.43-4549 VERBOSE=y test
> \
> TSPLUGINS_PATH=/build/source/bin/release-x86_64-localhost LD_LIBRARY_PATH=/build/source/bin/release-x86_64-localhost \
> TS_CURL_RETRY="RETRY=5,INTERVAL=100,HOST=tsduck.io" \
> /build/source/bin/release-x86_64-localhost/utest 
> *** MemoryPluginTest::All, incorrect value, utestMemoryPlugin.cpp, line 209
> expected: 3 ("REF_PACKETS_COUNT")
> actual:   0 ("output_packets.size()")
> *** MessageQueueTest::Queue, assertion failure, utestMessageQueue.cpp, line 145
> condition: thread.start()
> *** ThreadTest::DeleteWhenTerminated, assertion failure, utestThread.cpp, line 190
> condition: thread->start()
> *** ThreadTest::MutexTimeout, assertion failure, utestThread.cpp, line 258
> condition: thread.start()
> *** ThreadTest::Termination, assertion failure, utestThread.cpp, line 141
> condition: thread.start()
> *** TSProcessorTest::Processing, incorrect value, utestTSProcessor.cpp, line 230
> expected: 5 ("5")
> actual:   1 ("handler1.logs.size()")
> *** FAILURES (6 tests FAILED, 682 passed, 6 assertions FAILED, 32561 passed)
> make[2]: *** [Makefile:45: test-shared] Error 1
> make[1]: *** [Makefile:42: test] Error 2
> make: *** [Makefile:29: test] Error 2
> error: Cannot build '/nix/store/hh3i55xk8fqm2pl03nmv7n6jfswradj8-tsduck-3.43-4549.drv'.
> ```

I'm baffled. It builds on every major system, I just posted my results above.

TBH I don't know what `condition: thread.start()` means as a test failure, precisely ... @SuperSandro2000  could it be your build ran into cgroups / ulimit or OOM somehow?

tbutter commented:
> MessageQueueTest



> > The tests are failing:
> > ```
> > buildPhase completed in 7 minutes 21 seconds
> > Running phase: checkPhase
> > check flags: -j24 SHELL=/nix/store/v8sa6r6q037ihghxfbwzjj4p59v2x0pv-bash-5.3p9/bin/bash NODEKTEC=1 NOGITHUB=1 NOHIDES=1 NOPCSC=1 NOVATEK=1 SYSPREFIX=/ SYSROOT=/nix/store/sfd2ffwiz02c486mbb5v0cv2m5h0y1ha-tsduck-3.43-4549 VERBOSE=y test
> > \
> > TSPLUGINS_PATH=/build/source/bin/release-x86_64-localhost LD_LIBRARY_PATH=/build/source/bin/release-x86_64-localhost \
> > TS_CURL_RETRY="RETRY=5,INTERVAL=100,HOST=tsduck.io" \
> > /build/source/bin/release-x86_64-localhost/utest 
> > *** MemoryPluginTest::All, incorrect value, utestMemoryPlugin.cpp, line 209
> > expected: 3 ("REF_PACKETS_COUNT")
> > actual:   0 ("output_packets.size()")
> > *** MessageQueueTest::Queue, assertion failure, utestMessageQueue.cpp, line 145
> > condition: thread.start()
> > *** ThreadTest::DeleteWhenTerminated, assertion failure, utestThread.cpp, line 190
> > condition: thread->start()
> > *** ThreadTest::MutexTimeout, assertion failure, utestThread.cpp, line 258
> > condition: thread.start()
> > *** ThreadTest::Termination, assertion failure, utestThread.cpp, line 141
> > condition: thread.start()
> > *** TSProcessorTest::Processing, incorrect value, utestTSProcessor.cpp, line 230
> > expected: 5 ("5")
> > actual:   1 ("handler1.logs.size()")
> > *** FAILURES (6 tests FAILED, 682 passed, 6 assertions FAILED, 32561 passed)
> > make[2]: *** [Makefile:45: test-shared] Error 1
> > make[1]: *** [Makefile:42: test] Error 2
> > make: *** [Makefile:29: test] Error 2
> > error: Cannot build '/nix/store/hh3i55xk8fqm2pl03nmv7n6jfswradj8-tsduck-3.43-4549.drv'.
> > ```
> 
> I'm baffled. It builds on every major system, I just posted my results above.
> 
> TBH I don't know what `condition: thread.start()` means as a test failure, precisely ... @SuperSandro2000 could it be your build ran into cgroups / ulimit or OOM somehow?

I think after the first failure the test framework shuts down and does not allow creating new threads. The tests do an asset on thread.start() and fail.

https://github.com/tsduck/tsduck/blob/de67425c97a9735195da7b78b895e4985af085f2/src/utest/utestTSUnitThread.h

siriobalmelli committed:
tsduck: fall back to SCHED_OTHER

ofborg builders can run under Linux scheduler policies that
pthread_attr_setschedpolicy rejects.

Retry with SCHED_OTHER so TSDuck does not fail all
thread-dependent tests before they start.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

tbutter reviewed (approved):
built locally on x86_64-linux and tried the binaries.

looks good to me.

siriobalmelli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 454832`
Commit: `88e642034e63a0d3ee6c1a539ee33a487cbdd212`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>
